src/Treap/Pure.hs

```diff
@@ -82,6 +82,12 @@ data Treap m a
     deriving stock (Show, Read, Eq, Generic, Foldable)
     deriving anyclass (NFData)
 
+instance Measured m a => Semigroup (Treap m a) where
```

Comment:
Let's add documentation to this instance saying that `<>` is implemented via `merge`.